src/v/cluster/persisted_stm.cc

```diff
@@ -561,8 +554,9 @@ ss::future<> persisted_stm<T>::start() {
             _insync_offset = model::prev_offset(offset);
             set_next(offset);
         }
-        _resolved_when_snapshot_hydrated.set_value();
     }
+    _on_snapshot_hydrated.broadcast();
+    _snapshot_hydrated = true;
```

Comment:
Do we need to set this before the broadcast? The broadcast seems to check the predicate based on my reading of the seastar code.

Refs:

https://github.com/scylladb/seastar/blob/6e80e84a3c3cfb2daa5ec0ab37ec264bbc05081f/include/seastar/core/condition-variable.hh#L162-L163
https://github.com/scylladb/seastar/blob/6e80e84a3c3cfb2daa5ec0ab37ec264bbc05081f/src/core/condition-variable.cc#L92

Reply:
yes, great catch